### PR TITLE
Use processmon

### PR DESCRIPTION
**Pull request by thijsc**
Use `processmon` in skeleton template and a number of apps. I haven't been able to keep up with the changes, so I propose we merge this as is and port other apps separately.

### CODE REVIEW

**Review by shairyar — APPROVED** [review]

**Comment by tombruijn** [comment]
I think you need to remove the `support/processmon/processmon` file because it's a checked in file and when you run `rake global:install_processmon` it will write the executable to that location. Git will track the change and it can be accidentally be committed. Use a `.keep` file in the directory instead if you want to check the directory into git.

**Comment by thijsc** [comment]
> I think you need to remove the `support/processmon/processmon` file because it's a checked in file and when you run `rake global:install_processmon` it will write the executable to that location. Git will track the change and it can be accidentally be committed. Use a `.keep` file in the directory instead if you want to check the directory into git.

This was intentional, so people that don't have Rust installed can still use this repo.

**Review by jeffkreeftmeijer — APPROVED** [review]

**Comment by tombruijn** [comment]
> This was intentional, so people that don't have Rust installed can still use this repo.

But people don't have to have Rust installed. That's what the Dockerimage is for, right? They just have to run it once and it's installed.

**Comment by thijsc** [comment]
The docker image currently does not include procession. We'd have to set up an image layer and I haven't had time to figure out how that works.

**Comment by tombruijn** [comment]
Then I don't understand how this works.

This is what I see in this PR:

- We have a build environment to build processmon that everyone can build locally without having Rust installed because you it's build in Docker using `rake global:install_processmon`. It is build to `support/processmon/processmon`.
- On `rake app:up` the `support/processmon/processmon` file is copied to the app's `commands` directory so that it is available in the app's docker container after build.
- When `rake app:up` has built the app's Docker image, the `processmon` file is removed from the app's `commands` directory. (But it's still available on the Docker image.)

So we don't have to check in processmon necessarily because everyone can build it. It's a binary blob I'd prefer to leave out. We can build processmon on first boot if it's missing, and users can manually build a new version with `rake global:install_processmon`.

**Comment by thijsc** [comment]
@tombruijn your description of how it works is correct. The reason I added the binary is that it would be useful if customers can use this repo without jumping through hoops.

**Comment by tombruijn** [comment]
@thijsc They don't have to jump through hoops. We can automate it. It only takes a bit more time the very first time users run it.

**Comment by thijsc** [comment]
I don't have a strong opinion here, just explaining my thinking. If we want to take a different approach that's fine with me :-)

**Comment by thijsc** [comment]
@tombruijn do you think we should make the change you propose before merging?

**Comment by tombruijn** [comment]
@thijsc yes, otherwise we have the 5MB binary in the history

File: nodejs/next-js/commands/prepare [hunk 1]
```diff
@@ -1,15 +1,12 @@
-#!/bin/sh
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (tombruijn) [reply nodejs/next-js/commands/prepare:2]: fixed later on main

File: support/processmon/build.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
```
[the remark this answers is not in the document]
Comment (tombruijn) [reply support/processmon/build.sh:2]: fixed later on main

**Pull request merged**